azure/scope/machinepool.go

```diff
@@ -477,18 +486,18 @@ func (m *MachinePoolScope) createMachine(ctx context.Context, machine azure.VMSS
 					UID:                m.AzureMachinePool.UID,
 				},
 			},
-			Labels: map[string]string{
-				m.ClusterName():                 string(infrav1.ResourceLifecycleOwned),
-				clusterv1.ClusterNameLabel:      m.ClusterName(),
-				infrav1exp.MachinePoolNameLabel: m.AzureMachinePool.Name,
-			},
+			Annotations: map[string]string{},
 		},
 		Spec: infrav1exp.AzureMachinePoolMachineSpec{
 			ProviderID: machine.ProviderID(),
 			InstanceID: machine.InstanceID,
 		},
 	}
 
+	labels := m.getMachinePoolMachineLabels()
+	labels[m.ClusterName()] = string(infrav1.ResourceLifecycleOwned)
```

Comment:
why is this label not part of the `getMachinePoolMachineLabels` function?

Reply:
I meant for `getMachinePoolMachineLabels` to be a function for getting the labels we want to select AMPMs on. But maybe we could include that in the selector as well if it's present on every machine.